src/engines/arps/ARP/audio.cpp

```diff
@@ -81,69 +90,318 @@ namespace otto::engines::arp {
   }
 
   // PlayMode functions
-  void up(ArpeggiatorState& state, NoteArray& notes, std::uint8_t& current_step, OctaveModeFunc oct_func, NoteVector& output)
+  auto get_next_upwards = [](NoteArray& notes, std::uint8_t& current_step)
```

Comment:
I see no reason for these to be lambdas. It's probably alright, but i think they might as well just be regular functions...